### ISSUE TITLE
Input lag on HiDPI 4K when maximizing window

### ISSUE THREAD

**Issue by clamydo** [wez/wezterm#4110]
### What Operating System(s) are you seeing this problem on?

Windows

### Which Wayland compositor or X11 Window manager(s) are you using?

Windows 10

### WezTerm version

20230712-072601-f4abf8fd

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

No, and I'll explain why below

### Describe the bug

I experience a noticeable input lag (up to estimated 500 ms) proportional to the size of the window and of the screen resolution (when I move a window between displays of different resolutions). This is true for both WebGPU and OpenGL mode. The performance on Dx12 is much better than with Vulkan backend.

However, it *feels* to me, in fullscreen (without decorator) the lag is smaller again even on 4K resolution.

I am on a Intel(R) Core(TM) i7-8565U CPU with integrated Intel GPU. Is it too weak?

I tried the current nightly (20230712-072601-f4abf8fd), but it crashed when resizing the window.

On my, admittedly substantially beefier, Linux machine, I have no issues.

### To Reproduce

Open a new window, type, observe lag between pressing key and appearance of the character on the screen.

### Configuration

```lua
config.front_end = "WebGpu"
config.webgpu_power_preference = "HighPerformance"
config.webgpu_preferred_adapter = {
	backend = "Dx12",
	device = 16032,
	device_type = "IntegratedGpu",
	name = "Intel(R) UHD Graphics 620",
	vendor = 32902,
}

```

### Expected Behavior

No (noticable) input lag, even on slightly dated laptop hardware.

### Logs

```
Debug Overlay
wezterm version: 20230712-072601-f4abf8fd x86_64-pc-windows-msvc
Window Environment: Windows
WebGPU: name=Intel(R) UHD Graphics 620, device_type=IntegratedGpu, backend=Dx12, vendor=32902, device=16032
Enter lua statements or expressions and hit Enter.
Press ESC or CTRL-D to exit
```

**Comment by wez** [comment]
FWIW, the resize crash is from the upstream wgpu crate:
* https://github.com/gfx-rs/wgpu/issues/3967

**Comment by clamydo** [comment]
> Are you saying that every key press takes 500ms to render?
Exactly. The time between key press and rendering. Note, that 500 ms an estimate - I can try to measure this with my slomo mode of my smartphone camera?

> Only in certain applications? Is it that bad in the debug overlay itself?
Independent of applications and shells (in WSL2!) and also in the debug overlay.

**Comment by MartenBE** [comment]
I am also having this issue. Maximizing the window on 4K screen really increases the CPU usage used by wezterm and it becomes slow to respond. Resize it to a smaller window and cpu usage and lag also scale down.
